pyOCD/flash/flash.py

```diff
@@ -70,16 +70,22 @@ class Flash(object):
     def __init__(self, target, flash_algo):
         self.target = target
         self.flash_algo = flash_algo
-        self.end_flash_algo = flash_algo['load_address'] + len(flash_algo)*4
-        self.begin_stack = flash_algo['begin_stack']
-        self.begin_data = flash_algo['begin_data']
-        self.static_base = flash_algo['static_base']
-        self.page_size = flash_algo['page_size']
+        if flash_algo is not None:
+            self.end_flash_algo = flash_algo['load_address'] + len(flash_algo)*4
+            self.begin_stack = flash_algo['begin_stack']
+            self.begin_data = flash_algo['begin_data']
+            self.static_base = flash_algo['static_base']
+            self.page_size = flash_algo['page_size']
 
     def init(self):
         """
         Download the flash algorithm in RAM
         """
+
+        if self.flash_algo is None:
```

Comment:
Shall this logging error be only in __init__ method ? It's all over this file, as it can be once there? 

in __init__:

```
if flash_algo is not None:
   #set here all those flash_algo
else:
   error("Target does not have a flash algo")
```